cmd/serverless/main.go

```diff
@@ -244,136 +249,68 @@ func runAgent(stopCh chan struct{}) (daemon *serverless.Daemon, err error) {
 		log.Error("No API key configured, exiting")
 	}
 
-	// starts logs collection
-	// ----------------------
+	logChannel := make(chan *logConfig.ChannelMessage)
 
-	// type of logs we are subscribing to
-	var logsType []string
-	if envLogsType, exists := os.LookupEnv(logsLogsTypeSubscribed); exists {
-		parts := strings.Split(strings.TrimSpace(envLogsType), " ")
-		for _, part := range parts {
-			part = strings.ToLower(strings.TrimSpace(part))
-			switch part {
-			case "function", "platform", "extension":
-				logsType = append(logsType, part)
-			default:
-				log.Warn("While subscribing to logs, unknown log type", part)
-			}
-		}
-	} else {
-		logsType = append(logsType, "platform", "function", "extension")
-	}
+	forwarderTimeout := config.Datadog.GetDuration("forwarder_timeout") * time.Second
+	metricAgent := &metrics.ServerlessMetricAgent{}
+	metricAgent.Start(forwarderTimeout, &metrics.MetricConfig{}, &metrics.MetricDogStatsD{})
+	serverlessDaemon.SetStatsdServer(metricAgent)
+	serverlessDaemon.SetMuxHandle(logsAPICollectionRoute, logChannel, config.Datadog.GetBool("serverless.logs_enabled"), config.Datadog.GetBool("enhanced_metrics"))
```

Comment:
From this name I'm not really sure what this function is doing. Setting a mutex handle?

Reply:
I've renamed it 👍